src/main/java/screens/DisplayListingScreen.java

```diff
@@ -0,0 +1,119 @@
+package screens;
+
+import intefaceAdapters.displayListing.DisplayListingController;
+import intefaceAdapters.displayListing.DisplayListingResponseFormatter;
+import intefaceAdapters.displayListing.DisplayListingScreenInterface;
+import useCases.displayListing.DisplayListingDsRequestModel;
+import useCases.displayListing.DisplayListingPresenter;
+import useCases.displayListing.DisplayListingResponseModel;
+
+import javax.swing.*;
+import javax.swing.text.JTextComponent;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.List;
+
+public class DisplayListingScreen extends JPanel implements ActionListener, DisplayListingScreenInterface {
+
+    private final JTextField brand = new JTextField(25);
+    private final JTextField name = new JTextField(25);
+    private final JTextField color = new JTextField(25);
+    private final JTextField maxYear = new JTextField(25);
+    private final JTextField minYear = new JTextField(25);
+    private final String[] seats = {"2", "3", "4", "5", "6", "7",  "8"};
+    private final JComboBox<String> seatList = new JComboBox<>(seats);
+    private final JTextField maxPrice = new JTextField(25);
+    private final JTextField minPrice = new JTextField(25);
+
+    private final JTextArea description = new JTextArea(5, 20);
+    private final String[] types = {"Used", "New", "Rent"};
+    private final JComboBox<String> typeList = new JComboBox<>(types);
+    public final JButton back;
+
+    private final DisplayListingController displayListingController;
+
+    public DisplayListingScreen() {
```

Comment:
This code for this screen is massive and hard to read. I suggest taking the part where you build the components out into a separate class or helper method to make the code easier to read.

Reply:
will do that, thanks for the great rec!